src/main/java/note/NoteList.java

```diff
@@ -124,38 +126,55 @@ public void createNoteProcess(SceneList sceneList) {
         createNote(newNote);
     }
 
-    public void openNoteProcess() {
-        boolean checkExistance = ui.printOpenNoteMessage(this);
-        if (checkExistance) {
+    public void openNoteProcess() throws InvalidNoteException {
+        boolean checkExistence = ui.printOpenNoteMessage(this);
+        if (checkExistence) {
             String userInput = ui.readUserInput();
             while (!userInput.equals("")) {
-                if (!userInput.equals("search") && !userInput.equals("open")) {
+                if (!userInput.startsWith("search") && !userInput.startsWith("open")) {
                     ui.printNoteErrorMessage(INVALID_NOTE_COMMAND_MESSAGE);
```

Comment:
I think can do like isValid = !userInput.startsWith("search") && !userInput.startsWith("open") then in the if statement use if(isValid)